test/MSISonar_TEST.cc

```diff
@@ -49,7 +58,12 @@ class Sonar_TEST: public RenderingFixture
     obj->setVisibilityFlags(GZ_VISIBILITY_ALL);
     sceneNode->attachObject(obj);
     sceneNode->setVisible(true);
+
+#if GAZEBO_MAJOR_VERSION >=8
+    sceneNode->setPosition(_pose.X(),_pose.Y(),_pose.Z());
+#else
     sceneNode->setPosition(_pose.x,_pose.y,_pose.z);
+#endif
```

Comment:
I think it would be easier to make `_pose` to be an `ignition::math::Vector3`. It will avoid all the `#if GAZEBO_MAJOR_VERSION >=8` checks in this file.